File: March 2023/Day 11/Yet another query problem/yet-another-query-problem.py
from typing import List



class Solution:
    # Time Complexity :- O(n^2);
    # Space Complexity :- O(n);
    def solveQueries(self, N : int, num : int, A : List[int], Q : List[List[int]]) -> List[int]:
        # code here
        fre = [0]*N
        for i in range(0,N) :
            for j in range(i,N) :
                if A[i]==A[j] :
                    fre[i]+=1
        ans = []
        for q in Q :
            f=0
            for i in range(q[0],q[1]+1) :
                if fre[i]==q[2] :
                    f+=1
            ans.append(f)
        return ans



#{ 
 # Driver Code Starts
class IntArray:
    def __init__(self) -> None:
        pass
    def Input(self,n):
        arr=[int(i) for i in input().strip().split()]#array input
        return arr
    def Print(self,arr):
        for i in arr:
            print(i,end=" ")
        print()



class IntMatrix:
    def __init__(self) -> None:
        pass
    def Input(self,n,m):
        matrix=[]
        #matrix input
        for _ in range(n):
            matrix.append([int(i) for i in input().strip().split()])
        return matrix
    def Print(self,arr):
        for i in arr:
            for j in i:
                print(j,end=" ")
            print()


if __name__=="__main__":
    t = int(input())
    for _ in range(t):
        
        N = int(input())
        
        
        num = int(input())
        
        
        A=IntArray().Input(N)
        
        
        Q=IntMatrix().Input(num, 3)
        
        obj = Solution()
        res = obj.solveQueries(N, num, A, Q)
        
        IntArray().Print(res)
        

# } Driver Code Ends

File: March 2023/Day 11/Yet another query problem/yet-another-query-problem.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    // Time Complexity :- O(n^2);
    // Space Complexity :- O(n^2);
    vector<int> solveQueries(int N, int num, vector<int> &A, vector<vector<int>> &Q) {
        // code here
        vector<int> fre(N,0);
        for(int i=0;i<N;i++) {
            for(int j=i;j<N;j++) if(A[i]==A[j]) fre[i]++;
        }
        vector<int> list;
        for(vector<int> q : Q) {
            int f=0;
            for(int i=q[0];i<=q[1];i++) {
                if(fre[i]==q[2]) f++;
            }
            list.push_back(f);
        }
        return list;
    }
};


//{ Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        int N;
        cin>>N;
        int num;
        cin>>num;
        vector<int> A(N);
        for(int i=0;i<N;i++){
            cin>>A[i];
        }
        vector<vector<int>> Q(num, vector<int>(3));
        for(int i=0;i<num;i++){
            for(int j=0;j<3;j++){
                cin>>Q[i][j];
            }
        }
        Solution obj;
        vector<int> res = obj.solveQueries(N, num, A, Q);
        
        for(auto ele:res){
            cout<<ele<<" ";
        }
        cout<<endl;
    }
}

// } Driver Code Ends